src/mjolnir/restrictionbuilder.cc

```diff
@@ -331,8 +329,7 @@ void build(const std::string& complex_restriction_from_file,
             if (restriction.type() < RestrictionType::kOnlyRightTurn ||
                 restriction.type() > RestrictionType::kOnlyStraightOn) {
 
-              GraphId currentNode = GraphId(tile->id().tileid(), tile->id().level(), i);
-              GraphId tileid = tile->id();
+              GraphId currentNode = directededge.endnode();
```

Comment:
Turned out that both forward and it's opposite edge are marked with `start_restriction` flag.
So let's say we have a restriction `A - B - C`. Both `AB` & `BA` are `start_restriction`-s.
Before this change:
- algorithm starts with the node A and succeeds when `directededge == AB`
After this change
- algorithm starts with the node A and succeeds when `directededge == BA`

so the PR doesn't breaks anything